### DISCUSSION TITLE
Indexing of custom correlation ID

### DISCUSSION THREAD

**Discussion by R4cOOn** [datalust/seq-tickets#1901] [Help] [answered]
Hi,

We have a custom property `@Properties['X-Correlation-Id']` that we use to track request through multiple methods and apps.   
When getting "older" logs (that aren't in the RAM/cache) I can see that Seq goes through all events (as the status at the bottom of the query results show) and it takes a really long time (10-15min for a couple day old event).

According to the indexing documentation at https://docs.datalust.co/docs/indexing adding a signal should create an index and speed things up.  
That works fine for built-in signals (like errors or warnings that take seconds to show up even for events that ar e a week old).  
I added a new signal with column `@Properties['X-Correlation-Id']` and query `@Properties['X-Correlation-Id'] != null` but that doesn't speed anything up :(

Does that mean that indexing realies on a "fixed" value to work?  
i.e. having a signal `@Properties['X-Correlation-Id'] = '123'` will work but it cannot index a "column".

Is there any way to improve performance for my use case (speed up querying for my custom tracing property)?

Cheers.

**Comment by KodrAus** [comment, marked as answer]
Hi @R4cOOn :wave:

Seq's existing indexing scheme is based on predicates, which is good for low-cardinality properties like `Environment = 'Production'` or `@Level = 'Warning'`, but isn't suitable for high-cardinality properties like finding all events with a particular correlation id. We've been talking for a while about introducing some high-cardinality indexing, which is also something we'd need to properly support tracing. It doesn't have a locked-in design just yet or a timeline, but is definitely high on the list of things we want to do.
